Userland/Services/RequestServer/ConnectionCache.h

```diff
@@ -360,7 +369,10 @@ void start_connection(URL::URL const& url, auto job, auto& sockets_for_url, size
                 });
             } else {
                 cache.with_write_locked([&](auto&) {
-                    dbgln_if(REQUESTSERVER_DEBUG, "Immediately start request for url {} in {} - {}", url, &connection, connection.socket.ptr());
+                    connection.socket.with_read_locked([&](auto const& socket) {
```

Comment:
at a cursory glance, my question with this PR is:

Are we consistent with the order we take these locks in?

There seems to be several RW locks in the data structures here, and we should be mindful of lock ordering and potential deadlocks.

Reply:
er. hold up. this read lock take is only for the dbgln?

In that case, we should be guarding these as 
```
if constexpr (REQUESTSERVER_DEBUG) {
    foo.with_read_locked( ... ) {
        dbgln("we took da lock and are looking at information");
    }
}
```
so as to not take the lock unnecessarily when not debugging.

Reply:
Thanks for reviewing. I've made all the debugging locks only happen with REQUESTSERVER_DEBUG enabled.

I've removed that socket read lock to outside the cache write lock. The other socket write lock does need to be within the cache write lock. I don't think this should cause any issues? The socket lock doesn't depend on the cache lock anywhere else.

I'm not super experienced with this style of multithreading, so It’s very possible I’m missing something. Testing with a lot of requests, I haven’t run into a deadlock yet.